frontend/packages/dev-console/src/utils/add-resources-menu-utils.ts

```diff
@@ -0,0 +1,45 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { ImportOptions } from '../components/import/import-types';
+
+const PART_OF = 'app.kubernetes.io/part-of';
+
+export const getAddPageUrl = (
+  obj: K8sResourceKind,
+  type: string,
+  hasApplication: boolean,
+): string => {
+  let pageUrl = '';
+  const {
+    metadata: { namespace: ns },
+  } = obj;
+  switch (type) {
+    case ImportOptions.GIT:
+      pageUrl = `/import/ns/${ns}?importType=git`;
+      break;
+    case ImportOptions.CONTAINER:
+      pageUrl = `/deploy-image/ns/${ns}?preselected-ns=${ns}`;
+      break;
+    case ImportOptions.CATALOG:
+      pageUrl = '/catalog';
+      break;
+    case ImportOptions.DOCKERFILE:
+      pageUrl = `/import/ns/${ns}?importType=docker`;
+      break;
+    case ImportOptions.DATABASE:
+      pageUrl = `/catalog/ns/${ns}?category=databases`;
+      break;
+    default:
+      pageUrl = `add/ns/${ns}`;
+  }
+  const seperator = pageUrl.includes('?') ? '&' : '?';
```

Comment:
You could use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to manage all your attributes. Create it at the start and just `.append` to it in some case statements and let it do it's magic. Then just `searchParams.toString()` at the end to get the output:
`searchParams.toString(); // "q=URLUtils.searchParams&topic=More+webdev"`